Implemented Checkboxes Multi/Single-Select Field

miguelaenlle opened the pull request:
https://github.com/user-attachments/assets/17410aae-4d1f-4926-b596-f7cbb1576b91

- Implemented `Checkboxes` component with multi-select and single-select capabilities, based on the `Checkboxes` component used in the HackIllinois 2024 Site

- Checkbox component is responsive to different screen sizes (though not mobile-optimized just yet):

<img width="329" alt="Screenshot 2024-10-25 at 11 06 57 PM" src="https://github.com/user-attachments/assets/cc47ac94-6b04-4a8a-b343-fe56d4684de2">

- Checkbox component implements `useFormContext`, so it functions as any other form component

- Checkbox component lets users type in "Other" responses:

![image](https://github.com/user-attachments/assets/102059ed-2a14-4ae1-a92d-a3bcc5ce24be)

- Created temporary `TestForm` component to test the `Checkboxes` component with multiple questions, screen sizes, and to test the `useFormContext` functionality.

miguelaenlle committed:
Implemented Checkbox multi-select and single-select field

miguelaenlle committed:
Merge branch '2025-dev' into ma/single-and-multi-select-button

J164 commented on components/Form/Checkboxes/Checkboxes.tsx:
```diff
@@ -0,0 +1,182 @@
+
+    // `isRadioButton: true` indicates that this specific checkbox is a radio button
+    // and the css will change accordingly
+    isRadioButton?: boolean;
```
What is the difference between isRadio and isRadioButton?

miguelaenlle replied:
There's no difference -- this was part of the original Checkboxes component from site-2024. I'll merge this and isRadio

J164 commented on components/Form/Checkboxes/Checkboxes.tsx:
```diff
@@ -0,0 +1,182 @@
+    // there can be exactly one checkbox that serves as an "Other" option, and will allow the user
+    // to enter their own choice for the `value` through a text input
+    isOther?: boolean;
+    otherPlaceholder?: string;
```
Could we adjust this type so the placeholder can only be specified if isOther is true?

miguelaenlle replied:
I'll add that

J164 reviewed (changes requested):
This is a great start! Two questions: 
How can we make a CheckBox option required?
Can we make it so the Other text field is required when Other is selected?

miguelaenlle committed:
Merged isRadio and isRadioButton; adjusted CheckboxOption type so otherplaceholder is an option only if isRadio is true

miguelaenlle committed:
Implemented required functionality for checkboxes and required 'Other' input

miguelaenlle commented:
> This is a great start! Two questions: 
> How can we make a CheckBox option required? 
> Can we make it so the Other text field is required when Other is selected?

https://github.com/user-attachments/assets/13d8c486-a18f-4aa2-9dc0-11aaa70d1c04

I added these functionalities in the last commit:
- Ability to make CheckBox required
- Other textfield is required when Other is selected (this is true for both if the whole CheckBox component is required or not)

- Removed isRadioButton from CheckboxOption Type
- Adjusted CheckboxOption Type such that otherPlaceholder can only be specified if isOther is true

J164 reviewed (approved):
Looks good! Thanks for the changes!
